### PR TITLE
Jetpack: Redirect invalid routes under jetpack/new to jetpack/connect

### PR DESCRIPTION
**Pull request by seear**
Any Calypso route `/jetpack/new/*` is invalid, so redirect to `/jetpack/connect`.

Preparation for whitelisting the `/jetpack/new` route: pMz3w-7v3-p2, /cc @rickybanister @richardmuscat 

**To Test**
* Checkout this brach
* http://calypso.localhost:3000/jetpack/new should show the 'add new site page'
* also query args should be allowed, like http://calypso.localhost:3000/jetpack/new?ref=calypso-selector
* Anything under /new/, for example http://calypso.localhost:3000/jetpack/new/invalid, should redirect to `/jetpack/connect`

### CODE REVIEW

File: client/jetpack-connect/index.js [hunk 1]
```diff
@@ -72,4 +72,5 @@ export default function() {
 	page( '/jetpack/sso/:siteId?/:ssoNonce?', controller.sso );
 	page( '/jetpack/sso/*', controller.sso );
 	page( '/jetpack/new', controller.newSite );
+	page( '/jetpack/new/*', controller.notFoundError );
```
Comment (tyxla) [inline client/jetpack-connect/index.js]: What happens for `/jetpack/new/*` will also happen for `/jetpack/bla` too - so I'd suggest having this route match any previously unmatched `/jetpack/*` - what do you think?

File: client/jetpack-connect/controller.js [hunk 2]
```diff
@@ -252,4 +253,15 @@ export default {
...
+
+	notFoundError( context, next ) {
+		const emptyContent = (
+			<EmptyContent
+				illustration="/calypso/images/illustrations/illustration-404.svg"
+				title={ translate( 'Uh oh. Page not found.' ) }
+				line={ translate( 'Sorry, the page you were looking for doesn\'t exist or has been moved.' ) } />
+		);
+		renderPage( context, emptyContent );
+		next();
```
Comment (tyxla) [inline client/jetpack-connect/controller.js]: We don't have a next callback, so `next` can can be safely removed.

**Comment by richardmuscat** [comment]
Nice.

Rather than a 404 style page I think it'd be preferable to just send folks to /jetpack/connect/

**Comment by seear** [comment]
> Rather than a 404 style page I think it'd be preferable to just send folks to /jetpack/connect/

Yeah, sounds good to me. I'll modify this PR to do that.

File: client/jetpack-connect/index.js (hunk 1 above)
Comment (seear) [inline client/jetpack-connect/index.js]: > What happens for /jetpack/new/* will also happen for /jetpack/bla

Yes, I agree, but in the short-term `/jetpack/bla` will still redirect to jetpack.com even after we add `jetpack/new/*` to the whitelist. Let's keep this PR for the `/new` routing problem (which is a current bug), and then plan how we want the route handling to work for all /jetpack/* routes.

Comment (tyxla) [inline client/jetpack-connect/index.js]: Sounds good to me.

**Comment by seear** [comment]
Changed to a redirect. This could use another look before merge.

**Comment by tyxla** [comment]
👀

**Pull request merged**